### PR TITLE
Create Tags endpoints

### PR DESCRIPTION
**Pull request by emilymdubois**
Refs https://github.com/osmlab/to-fix-backend/issues/140, https://github.com/osmlab/to-fix-backend/issues/156

This PR does the following:

- [x] Adds `metadata` to tags
- [x] Adds `GET /project/:project/tags` endpoint
- [x] Adds `POST /project/:project/tags` endpoint
- [x] Adds `GET /project/:project/tags/:tag` endpoint
- [x] Adds `PUST /project/:project/tags/:tag` endpoint
- [x] Adds `DELETE /project/:project/tags/:tag` endpoint
- [x] Adds `GET /project/:project/items/:item/tags` endpoint
- [x] Adds `POST /projects/:project/items/:items/tags` endpoint
- [x] Adds `DELETE /projects/:project/items/:item/tags/:tag` endpoint
- [x] Adds tests for all ☝️ endpoints
- [x] Ensure documentation is up-to-date and accurate

cc @batpad

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -48,6 +48,61 @@ curl https://host/projects
 ]
 ```
 
+### get-project-tag
```
Comment (samanpwbb) [inline API.md]: get we rename this to get-project-tags, since it returns a list of tags?

File: API.md [hunk 2]
```diff
@@ -48,6 +48,61 @@ curl https://host/projects
...
+    project_id: '00000000-0000-0000-0000-000000000000',
+    name: 'My Tag',
+    metadata: {},
+    createdAt: '2017-10-20T00:00:00.000Z',
+    updatedAt: '2017-10-20T00:00:00.000Z'
+  }
+]
+```
+
+### get-project-tag
```
Comment (samanpwbb) [inline API.md]: Is this endpoint necessary?

File: API.md (hunk 1 above)
Comment (emilymdubois) [reply API.md]: 👍 Will do!

File: API.md (hunk 2 above)
Comment (emilymdubois) [reply API.md]: Yeah, probably not! I think I'll keep it in for now while we get tags and comments into `master`, and then we can do a sweep after the UI is hooked up for clean-up and standardization. Does that sound good?

**Pull request merged**